shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/data/remote/datasource/AuthRemoteDataSource.kt

```diff
@@ -1,26 +1,20 @@
 package vn.luongvo.kmm.survey.data.remote.datasource
 
-import io.ktor.client.request.*
-import io.ktor.http.*
-import kotlinx.coroutines.flow.Flow
-import vn.luongvo.kmm.survey.data.remote.apiclient.ApiClient
-import vn.luongvo.kmm.survey.data.remote.apiclient.builder.path
-import vn.luongvo.kmm.survey.data.remote.body.LoginRequestBody
+import vn.luongvo.kmm.survey.data.remote.ApiClient
+import vn.luongvo.kmm.survey.data.remote.model.request.LoginRequestBody
+import vn.luongvo.kmm.survey.data.remote.model.response.TokenResponse
 
 interface AuthRemoteDataSource {
 
-    fun logIn(body: LoginRequestBody): Flow<String>
+    suspend fun logIn(body: LoginRequestBody): TokenResponse
 }
 
 class AuthRemoteDataSourceImpl(private val apiClient: ApiClient) : AuthRemoteDataSource {
 
-    override fun logIn(body: LoginRequestBody): Flow<String> {
-        return apiClient.body(
-            HttpRequestBuilder().apply {
-                path("/v1/oauth/token")
-                method = HttpMethod.Post
-                setBody(body)
-            }
+    override suspend fun logIn(body: LoginRequestBody): TokenResponse {
```

Comment:
Going forward, we will use `suspend` for repository? What happen to `flow`?

Reply:
@blyscuit suspend is like `async/await` and `flow` is more on `Combine` side, so I guest it's more related to the use case :D (e.g., we can use both `async/await` and `Combine`) :D

Reply:
We can use them together but it brings one more complexity to the project where `Repo = suspend` and `Usecase = flow`, suspend and flow can be used interchangeably, so I would like to know what's the limitation or decision to choose this pattern instead of using only `flow`.

Reply:
@blyscuit @suho `Flow` is better than suspended when we need to catch data & exceptions in a chain of requests with some built-in operators. I made use of `suspend` in DataSource layer to make it easy for exception throwing before catching and transforming it by `flowTransform` https://github.com/luongvo/kmm-survey/pull/50/files#diff-05c65290e168fa9ae695459bdd847e68dc7be1a7f2525db56b9a279145c1dacd. 

The problem with `suspended fun` is we need to catch errors by a `try/catch` block manually.